sentry-core/src/main/java/io/sentry/core/cache/CacheStrategy.java

```diff
@@ -0,0 +1,97 @@
+package io.sentry.core.cache;
+
+import static io.sentry.core.SentryLevel.ERROR;
+
+import io.sentry.core.ISerializer;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.util.Objects;
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import org.jetbrains.annotations.NotNull;
+
+abstract class CacheStrategy {
+
+  @SuppressWarnings("CharsetObjectCanBeUsed")
+  protected static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  protected final @NotNull SentryOptions options;
+  protected final @NotNull ISerializer serializer;
+  protected @NotNull File directory;
```

Comment:
`m`: This is currently read-only for `SessionCache` and `DiskCache`. I would enforce this by making `directory` private and creating a getter for it.

Reply:
they are `final` so read-only anyway